Mathlib/GroupTheory/GroupAction/Hom.lean

```diff
@@ -357,6 +358,61 @@ end MulActionHom
 
 end MulActionHom
 
+/-- Evaluation at a point as a `MulActionHom`. -/
+@[to_additive (attr := simps) "Evaluation at a point as an `AddActionHom`."]
+def Pi.evalMulActionHom {ι M : Type*} {X : ι → Type*} [∀ i, SMul M (X i)] (i : ι) :
+    (∀ i, X i) →[M] X i :=
+  ⟨Function.eval i, fun _ _ ↦ rfl⟩
+
+namespace MulActionHom
+
+section FstSnd
+
+variable {M α β : Type*} [SMul M α] [SMul M β]
+
+variable (M α β) in
+/-- `Prod.fst` as a bundled `MulActionHom`. -/
+@[to_additive (attr := simps -fullyApplied) "`Prod.fst` as a bundled `AddActionHom`."]
+def fst : α × β →[M] α where
+  toFun := Prod.fst
+  map_smul' _ _ := rfl
+
+variable (M α β) in
+/-- `Prod.snd` as a bundled `MulActionHom`. -/
+@[to_additive (attr := simps -fullyApplied) "`Prod.snd` as a bundled `AddActionHom`."]
+def snd : α × β →[M] β where
+  toFun := Prod.snd
+  map_smul' _ _ := rfl
+
+end FstSnd
+
+variable {M N α β γ δ : Type*} [SMul M α] [SMul M β] [SMul N γ] [SMul N δ] {σ : M → N}
+
+/-- If `f` and `g` are equivariant maps, then so is `x ↦ (f x, g x)`. -/
+@[to_additive (attr := simps -fullyApplied)
+  "If `f` and `g` are equivariant maps, then so is `x ↦ (f x, g x)`."]
+def prodMk (f : α →ₑ[σ] γ) (g : α →ₑ[σ] δ) : α →ₑ[σ] γ × δ where
```

Comment:
Looks like for `MonoidHom` and `RingHom` (at least) this map is called `prod`. Is there a reason that you used `prodMk` here? (I would find it more likely for `prodMk` to mean `x ↦ (x, x)`.)

Reply:
I think that `prodMk` is the right name, because it's about `Prod.mk` and `x ↦ (x, x)` should be called `diag`, but we should change it (or not, if other maintainers don't think so) for all bundled homs in 1 PR. Renamed to `prod` for now.